source/agora/common/Task.d

```diff
@@ -45,6 +48,7 @@ public class TaskManager
     public void runTask (void delegate() dg) nothrow
     {
         static import vibe.core.core;
+        this.tasks_started++;
```

Comment:
Well, is there any problem with the value overflow? I mean this value would exceed ulong.max.

Reply:
Create a simple D application:

```D
import std.stdio;

void main ()
{
    writeln(ulong.max);
    foreach (idx; 0 .. ulong.max)
    {
        if (idx % 100_000 == 0)
            writeln(idx);
    }
}
```

And run it.

Try to estimate how long it will take your computer to reach `ulong.max` and exit the `main()` function.

Reply:
> Try to estimate how long it will take your computer to reach ulong.max and exit the main() function.

I know how big it is like 2^64 or about 10^18. I was just curious about the situation. It's all right.

Reply:
According to https://blog.trailofbits.com/2019/11/27/64-bits-ought-to-be-enough-for-anybody/#:~:text=Modern%20processors%20operate%20on%2064,as%20a%20canonical%20impossible%20problem. it would take 132 years to reach ulong.max on a MacBook pro.

And this is if the only thing the CPU did was increment a counter at the maximum speed.

We will never reach ulong.max.

Reply:
> it would take 132 years to reach ulong.max on a MacBook pro.

It's good read it. But I think it's about brute force comparison for all 2^64 values. not just increment.